util/src/org/commcare/util/mocks/MockDataUtils.java

```diff
@@ -0,0 +1,143 @@
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.xmlpull.v1.XmlPullParserException;
+
+/**
+ * Methods that mostly are used around the mocks that replicate stuff from
+ * other projects.
+ *
+ * TODO: We should try to centralize how these are used.
+ *
+ * @author ctsims
+ */
+public class MockDataUtils {
+
+    public static MockUserDataSandbox getStaticStorage() {
+        LivePrototypeFactory factory = new LivePrototypeFactory();
+        PrototypeFactory.setStaticHasher(factory);
+        return new MockUserDataSandbox(factory);
+    }
+
+    public static void parseIntoSandbox(InputStream stream, MockUserDataSandbox sandbox) {
+        CommCareTransactionParserFactory factory = new CommCareTransactionParserFactory(sandbox);
+        try {
+            DataModelPullParser parser = new DataModelPullParser(stream, factory);
+            parser.parse();
+        } catch (IOException | UnfullfilledRequirementsException |
+                XmlPullParserException | InvalidStructureException ioe) {
+            ioe.printStackTrace();
+        }
+    }
+
+    /**
+     * For the users and groups in the provided sandbox, extracts out the list
+     * of valid "owners" for entities (cases, ledgers, etc) in the universe.
+     *
+     * Borrowed from Android implementation, should likely be centralized.
+     *
+     * TODO: Move this static functionality into CommCare generally.
+     */
+    public static Vector<String> extractEntityOwners(MockUserDataSandbox sandbox) {
+        Vector<String> owners = new Vector<String>();
+        Vector<String> users = new Vector<String>();
+
+        for (IStorageIterator<User> userIterator = sandbox.getUserStorage().iterate(); userIterator.hasMore(); ) {
+            String id = userIterator.nextRecord().getUniqueId();
+            owners.addElement(id);
+            users.addElement(id);
+        }
+
+        //Now add all of the relevant groups
+        //TODO: Wow. This is.... kind of megasketch
+        for (String userId : users) {
+            DataInstance instance = loadFixture(sandbox, "user-groups", userId);
+            if (instance == null) {
+                continue;
+            }
+            EvaluationContext ec = new EvaluationContext(instance);
+            for (TreeReference ref : ec.expandReference(XPathReference.getPathExpr("/groups/group/@id").getReference())) {
+                AbstractTreeElement<AbstractTreeElement> idelement = ec.resolveReference(ref);
+                if (idelement.getValue() != null) {
+                    owners.addElement(idelement.getValue().uncast().getString());
+                }
+            }
+        }
+
+        return owners;
+    }
+
+    /**
+     * Load the referenced fixture out of storage for the provided user
+     *
+     * @param sandbox
+     * @param refId
+     * @param userId
+     * @return
+     */
+    private static FormInstance loadFixture(MockUserDataSandbox sandbox,
```

Comment:
I prefer to not include argument param annotations unless they also have corresponding comments.